Ui/DataProvider/Menu/Form/MenuDataProvider.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Snowdog\Menu\Ui\DataProvider\Menu\Form;
+
+use Magento\Ui\DataProvider\AbstractDataProvider;
+use Snowdog\Menu\Model\ResourceModel\Menu\CollectionFactory;
+use Snowdog\Menu\Model\Menu;
+
+class MenuDataProvider extends AbstractDataProvider
+{
+    /** @var array */
+    private $loadedData = [];
+
+    public function __construct(
+        $name,
+        $primaryFieldName,
+        $requestFieldName,
+        CollectionFactory $collectionFactory,
+        array $meta = [],
+        array $data = []
+    ) {
+        $this->collection = $collectionFactory->create();
+        parent::__construct(
+            $name,
+            $primaryFieldName,
+            $requestFieldName,
+            $meta,
+            $data
+        );
+    }
+
+    public function getData(): array
+    {
+        if (!empty($this->loadedData)) {
```

Comment:
do not use `empty()` to check field is initialized, use `$this->loadedData !== null` instead (remove default property value of course at first)

Reply:
Changed